sphinxygen: init at 1.0.4

samueltardieu opened the pull request:
This package is needed to update `serd` (see #350876).

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

samueltardieu committed:
sphinxygen: init at 1.0.4

itepastra commented:
From the license text it seems like gitlab misinterpreted it, and the license is actually the [isc license](https://opensource.org/license/isc-license-txt)

samueltardieu commented:
> The serd library also uses the same isc license and that derivation containes the incorrect one as well

I've fixed this one in #351223 as well, I'll mark it as ready later when its dependencies are merged. I've also updated the license in #351220, from the same author.

Aleksanaa commented on pkgs/by-name/sp/sphinxygen/package.nix:
```diff
@@ -0,0 +1,33 @@
+
+  src = fetchFromGitLab {
+    owner = "drobilla";
+    repo = pname;
```
```suggestion
    repo = "sphinxygen";
```

Aleksanaa commented on pkgs/by-name/sp/sphinxygen/package.nix:
```diff
@@ -0,0 +1,33 @@
+
+  pythonImportsCheck = [ "sphinxygen" ];
+
+  meta = with lib; {
+    description = "Generates Sphinx markup from an XML description extracted by Doxygen";
+    homepage = "https://gitlab.com/drobilla/sphinxygen";
+    changelog = "https://gitlab.com/drobilla/sphinxygen/-/releases/v${version}";
+    license = licenses.isc;
+    maintainers = with maintainers; [ samueltardieu ];
```
```suggestion
  meta = {
    description = "Generates Sphinx markup from an XML description extracted by Doxygen";
    homepage = "https://gitlab.com/drobilla/sphinxygen";
    changelog = "https://gitlab.com/drobilla/sphinxygen/-/releases/v${version}";
    license = lib.licenses.isc;
    maintainers = with lib.maintainers; [ samueltardieu ];
```
